src/compiler/scala/tools/nsc/ast/TreeBrowsers.scala

```diff
@@ -683,6 +684,7 @@ object TreeBrowsers {
     def format(width: Int, writer: Writer): Unit = {
       type FmtState = (Int, Boolean, Document)
 
+      @tailrec
```

Comment:
BTW, would the pattern `case List()` two lines below be more efficient if it were `case Nil`?

Reply:
or `case _: Nil.type`. Where is the empty list posse?

Reply:
TIL `UnapplySeqWrapper`. Not sure whether to say "mind blown" or "code blown". Maybe this is why blandishments about `List()` are dangerous.